test/FastTests/Server/Documents/Queries/Dynamic/MapReduce/RavenDB_8076.cs

```diff
@@ -0,0 +1,214 @@
+﻿using Raven.Client.Documents.Queries;
+using Raven.Tests.Core.Utils.Entities;
+using Sparrow.Json;
+using Xunit;
+
+namespace FastTests.Server.Documents.Queries.Dynamic.MapReduce
+{
+    public class RavenDB_8076 : RavenTestBase
+    {
+        [Fact]
+        public void Can_sort_by_aggregation_function_in_dynamic_group_by_query()
+        {
+            using (var store = GetDocumentStore())
+            {
+                using (var s = store.OpenSession())
+                {
+                    s.Store(new Company
+                    {
+                        Address1 = "UK",
+                        AccountsReceivable = 1
+                    });
+
+                    s.Store(new Company
+                    {
+                        Address1 = "UK",
+                        AccountsReceivable = 1
+                    });
+
+                    s.Store(new Company
+                    {
+                        Address1 = "USA",
+                        AccountsReceivable = 1
+                    });
+
+                    s.SaveChanges();
+                }
+
+                using (var commands = store.Commands())
+                {
+                    foreach (var q in new(string Query, string Field)[]
+                    {
+                        (@"select count()
+                        from Companies
+                        group by Address1 order by count()", "Count"),
```

Comment:
We've been always storing the result of count operation in the auto map-reduce index as `Count` field. So if the alias isn't specified then we return it as it's stored in the index. We assign it here:
https://github.com/ravendb/ravendb/blob/v4.0/src/Raven.Server/Documents/Queries/QueryMetadata.cs#L263
It's rather just a convention than a requirement. I like it since `Count` looks better in the output than `count()`. Also if we have `select sum(Price) from Orders ...` we don't return `sum(Price)` but `Price`

Do you prefer to rename the default name from `Count` to `count()`?